src/NUnitEngine/nunit.engine/Services/DefaultTestRunnerFactory.cs

```diff
@@ -76,11 +76,13 @@ public override ITestEngineRunner MakeTestRunner(TestPackage package)
                     projectCount++;
             }
 
+            ProcessModel processModel = GetTargetProcessModel(package);
+
             // If we have multiple projects or a project plus assemblies
             // then defer to the AggregatingTestRunner, which will make
             // the decision on a file by file basis so that each project
             // runs with its own settings.
-            if (projectCount > 1 || projectCount > 0 && assemblyCount > 0)
+            if (projectCount > 1 || (projectCount > 0 && assemblyCount > 0) || (projectCount > 0 && processModel == ProcessModel.Default))
```

Comment:
The problem I  see is that when processmodel is not specified we actually don't know what runner we want to use for a project without looking inside it. That gives us two options:

1) Figure out a way to call the project service early enough to expand the project, so we can use that information to decide on the runner.
2) Assume that the AggregatingTestRunner will be able to figure it out, as you have done.

Can we take a look at whether option 1 is a possibility?